src/server/game/Spells/Auras/SpellAuras.cpp

```diff
@@ -851,6 +854,31 @@ void Aura::Update(uint32 diff, Unit* caster)
             }
         }
     }
+    
+    if (m_heartBeatResistTimer)
+    {
+        if (m_heartBeatResistTimer < int32(diff))
+        {     
+            m_heartBeatResistTimer += CalculatePct(m_maxDuration, 25) - diff;
```

Comment:
Can this cause `m_heartBeatResistTimer` to become zero? What happens if it does?

Reply:
If m_heartBeatResistTimer is equal to zero, the condition will not be met and we will get nothing. When the aura expires, will the timer continue to work? I think not, because class object will be deleted (correct me if I'm wrong).

Reply:
Can `diff` be equal to `m_heartBeatResistTimer` plus `CalculatePct(m_maxDuration, 25)`, for example if the server is heavily loaded?

If this happens, the new value for `m_heartBeatResistTimer` will be zero, causing it to not be processed from now on. Is that intentional?

Reply:
Well, I get that. Thank you.

Reply:
The value of `CalculatePct(m_maxDuration,25)` is between 250 and 15000.
Here the lowest `m_maxDuration `value of spell with attr `SPELL_ATTR0_HEARTBEAT_RESIST_CHECK `:

> ID - 35198 Terrify
=================================================
Description: Strikes fear in nearby enemies, causing them to flee in terror for $d.
ToolTip: Fleeing in terror.
Category = 0, SpellIconID = 98, activeIconID = 0, SpellVisual = (7518,0)
Family SPELLFAMILY_GENERIC, flag [0] 0x00000000 [1] 0x00000000 [2] 0x00000000
SpellSchoolMask = 32 (SPELL_SCHOOL_MASK_SHADOW)
DamageClass = 1 (SPELL_DAMAGE_CLASS_MAGIC)
PreventionType = 1 (SPELL_PREVENTION_TYPE_SILENCE)
=================================================
Attributes: 0x40010000 (SPELL_ATTR0_NOT_SHAPESHIFT, SPELL_ATTR0_HEARTBEAT_RESIST_CHECK)
AttributesEx1: 0x00000088 (SPELL_ATTR1_CANT_BE_REDIRECTED, SPELL_ATTR1_CANT_BE_REFLECTED)
=================================================
Spell Level = 0, base 0, max 0, maxTarget 0
Category = 0
DispelType = 0 (DISPEL_NONE)
Mechanic = 5 (MECHANIC_FEAR)
SpellRange: (Id 1) "Self Only":
    MinRange = 0, MinRangeFriendly = 0
    MaxRange = 0, MaxRangeFriendly = 0
CastingTime (Id 15) = 4,00
RecoveryTime: 0 ms, CategoryRecoveryTime: 0 ms
StartRecoveryCategory = 133, StartRecoveryTime = 1500,00 ms
Duration: ID (36)  1000, 0, 1000
Interrupt Flags: 0x0000000B, AuraIF 0x00000000, ChannelIF 0x00000000
Chance = 101, charges - 0

And higher:

> ID - 605 Mind Control
=================================================
Description: Controls a humanoid mind up to level $s1, but increases the time between its attacks by $s3%.  Lasts up to $d.
ToolTip: Charmed.  Time between attacks increased by $s3%.
Category = 0, SpellIconID = 235, activeIconID = 0, SpellVisual = (137,0)
Family SPELLFAMILY_PRIEST, flag [0] 0x00020000 [1] 0x00000000 [2] 0x00000000
SpellSchoolMask = 32 (SPELL_SCHOOL_MASK_SHADOW)
DamageClass = 1 (SPELL_DAMAGE_CLASS_MAGIC)
PreventionType = 1 (SPELL_PREVENTION_TYPE_SILENCE)
=================================================
Attributes: 0x40150000 (SPELL_ATTR0_NOT_SHAPESHIFT, SPELL_ATTR0_DONT_AFFECT_SHEATH_STATE, SPELL_ATTR0_STOP_ATTACK_TARGET, SPELL_ATTR0_HEARTBEAT_RESIST_CHECK)
AttributesEx1: 0x04022005 (SPELL_ATTR1_DISMISS_PET, SPELL_ATTR1_CHANNELED_1, SPELL_ATTR1_FARSIGHT, SPELL_ATTR1_UNAUTOCASTABLE_BY_PET, SPELL_ATTR1_UNK26)
AttributesEx2: 0x00080040 (SPELL_ATTR2_CANT_TARGET_TAPPED, SPELL_ATTR2_NOT_NEED_SHAPESHIFT)
AttributesEx4: 0x20000800 (SPELL_ATTR4_UNK11, SPELL_ATTR4_UNK29)
AttributesEx6: 0x00800000 (SPELL_ATTR6_UNK23)
=================================================
Creature Type Mask = 0x00000040 (HUMANOID)
Stances: FORM_SHADOW
Skill (Id 78) "Shadow Magic"
    ReqSkillValue 1, Forward Spell = 0, MinMaxValue (0, 0), CharacterPoints (0, 0)Spell Level = 30, base 30, max 0, maxTarget 0
Category = 0
DispelType = 1 (DISPEL_MAGIC)
Mechanic = 1 (MECHANIC_CHARM)
SpellRange: (Id 3) "Twenty yards":
    MinRange = 0, MinRangeFriendly = 0
    MaxRange = 20, MaxRangeFriendly = 20
CastingTime (Id 14) = 3,00
RecoveryTime: 0 ms, CategoryRecoveryTime: 0 ms
StartRecoveryCategory = 133, StartRecoveryTime = 1500,00 ms
Duration: ID (3)  60000, 0, 60000
Power POWER_MANA, Cost 12 %
Interrupt Flags: 0x0000000F, AuraIF 0x00000000, ChannelIF 0x00006C2C
Chance = 101, charges - 0

And we have some deprecated spells with `m_maxDuration `value -1:
```
31365 - Self Fear
31797 - Banish Self
32974 - Cinematic - Mind Control (Rank 1)
32976 - Cinematic - Mind Control (Rank 1)
38456 - Banish Self
```

If diff value is sufficient for this, we will get the corresponding behavior.